integration/compose/src/main/java/com/bumptech/glide/integration/compose/GlideModifier.kt

```diff
@@ -0,0 +1,252 @@
+import androidx.compose.ui.semantics.SemanticsPropertyReceiver
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.IntSize
+import com.bumptech.glide.RequestBuilder
+import com.bumptech.glide.integration.ktx.AsyncGlideSize
+import com.bumptech.glide.integration.ktx.ExperimentGlideFlows
+import com.bumptech.glide.integration.ktx.ImmediateGlideSize
+import com.bumptech.glide.integration.ktx.InternalGlideApi
+import com.bumptech.glide.integration.ktx.Placeholder
+import com.bumptech.glide.integration.ktx.ResolvableGlideSize
+import com.bumptech.glide.integration.ktx.Resource
+import com.bumptech.glide.integration.ktx.Status
+import com.bumptech.glide.integration.ktx.flowResolvable
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.plus
+import kotlin.math.roundToInt
+
+@ExperimentalGlideComposeApi
+@OptIn(InternalGlideApi::class)
+internal fun Modifier.glideNode(
+  requestBuilder: RequestBuilder<Drawable>,
+  size: ResolvableGlideSize,
+  contentScale: ContentScale,
+  alignment: Alignment,
+  colorFilter: ColorFilter?,
+): Modifier =
+  this then GlideNodeElement(requestBuilder, size, contentScale, alignment, colorFilter)
+
+@ExperimentalGlideComposeApi
+@OptIn(InternalGlideApi::class)
+internal data class GlideNodeElement constructor(
+  private val requestBuilder: RequestBuilder<Drawable>,
+  private val size: ResolvableGlideSize,
+  private val contentScale: ContentScale,
+  private val alignment: Alignment,
+  private val colorFilter: ColorFilter?,
+) : ModifierNodeElement<GlideNode>() {
+  override fun create(): GlideNode {
+    val result = GlideNode()
+    result.onNewRequest(requestBuilder, size, contentScale, alignment, colorFilter)
+    return result
+  }
+
+  override fun update(node: GlideNode) {
+    node.onNewRequest(requestBuilder, size, contentScale, alignment, colorFilter)
+  }
+}
+
+@ExperimentalGlideComposeApi
+@OptIn(InternalGlideApi::class)
+internal class GlideNode : DrawModifierNode, LayoutModifierNode, SemanticsModifierNode,
+  Modifier.Node() {
+  private var requestBuilder: RequestBuilder<Drawable>? = null
+  private lateinit var resolvableGlideSize: ResolvableGlideSize
+  private lateinit var contentScale: ContentScale
+  private lateinit var alignment: Alignment
+  private var colorFilter: ColorFilter? = null
+  private var currentJob: Job? = null
+  private var drawable: Drawable? = null
+
+  private var state: RequestState = RequestState.Loading
+
+  fun onNewRequest(
+    requestBuilder: RequestBuilder<Drawable>,
+    resolvableGlideSize: ResolvableGlideSize,
+    contentScale: ContentScale,
+    alignment: Alignment,
+    colorFilter: ColorFilter?,
+  ) {
+    this.requestBuilder = requestBuilder
+    this.resolvableGlideSize = resolvableGlideSize
+    this.contentScale = contentScale
+    this.alignment = alignment
+    this.colorFilter = colorFilter
```

Comment:
consider handling the case where alignment/colorfilter/etc change but the requestBuilder does not. In this case we can just invalidate draw or layout but not initiate an entirely new request

Reply:
Done